File: docs/w03-pid-control/code/p_margins.py
import control as ct

s = ct.tf("s")
G = 40 / (s * (s + 2) * (s + 10))   # pitch attitude / elevator (rad/rad)

Kp = 1.0
gm, pm, w180, wc = ct.margin(Kp * G)
print(f"GM = {gm:.2f} at {w180:.2f} rad/s, PM = {pm:.1f} deg at {wc:.2f} rad/s")

# tracking: status=draft version=0 assisted=true

File: docs/w03-pid-control/code/p_margins.m
s = tf('s');
G = 40 / (s*(s + 2)*(s + 10));   % pitch attitude / elevator (rad/rad)

Kp = 1;
[Gm, Pm, Wcg, Wcp] = margin(Kp*G);
fprintf('GM = %.2f at %.2f rad/s, PM = %.1f deg at %.2f rad/s\n', Gm, Wcg, Pm, Wcp)

% tracking: status=draft version=0
